atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/command/timestamp/CleanTransactionRange.java

```diff
@@ -85,14 +88,19 @@ protected int executeTimestampCommand(AtlasDbServices services) {
                     SafeArg.of("startTs", startTs), SafeArg.of("commitTs", commitTs));
 
             Cell key = Cell.create(rowName, TransactionConstants.COMMIT_TS_COLUMN);
-            toDelete.put(key, value.getTimestamp());  //value.getTimestamp() should always be 0L
+            byte[] valueForRollbackTimestamp =
+                    TransactionConstants.getValueForTimestamp(TransactionConstants.FAILED_COMMIT_TS);
+
+            txTableValuesToWrite.put(key, valueForRollbackTimestamp);
         }
 
-        if (!toDelete.isEmpty()) {
-            kvs.delete(TransactionConstants.TRANSACTION_TABLE, toDelete);
-            printer.info("Delete completed.");
+        if (!txTableValuesToWrite.isEmpty()) {
+            Map<TableReference, Map<Cell, byte[]>> valuesToPut = new HashMap<>();
+            valuesToPut.put(TransactionConstants.TRANSACTION_TABLE, txTableValuesToWrite);
+            kvs.multiPut(valuesToPut, AtlasDbConstants.TRANSACTION_TS);
```

Comment:
Don't think this will work because of impl details about the timestamp CAS uses. Should discuss offline with @ashrayjain and I.

Reply:
We should add a batch PUE API, I will add that in and then use it here. Lets hold off on the PR.

Reply:
does it need to be PUE or can it just be put? PUE has the additional overhead of paxos

Reply:
Also, Cassandra doesn't support a batch PUE across rows (only within the same row)

Reply:
put explicitly checks that its not the transaction table:

```
    @Override
    public void put(TableReference tableRef, Map<Cell, byte[]> values, long timestamp) throws KeyAlreadyExistsException {
        Validate.isTrue(timestamp != Long.MAX_VALUE);
        Validate.isTrue(timestamp >= 0);
        Validate.isTrue(!tableRef.equals(TransactionConstants.TRANSACTION_TABLE), TRANSACTION_ERROR);
        if (values.isEmpty()) {
            return;
        }
        delegate.put(tableRef, values, timestamp);
    }
```

I would propose adding a putToTransactionTable API for this, that bypasses the check and can write the ROLLBACK_TIMESTAMP at timestamp 0. Does that sound reasonable?

Reply:
writing at timestamp 0 is not reasonable as as that will again be will be hidden behind the cas' (which happen at wall clock time wrt cassandra)

Reply:
Yea that put needs to happen at wall clock time

Reply:
Should also probably do it at consistency ALL?

Reply:
I want the AtlasDb timestamp to be 0 but the Cassandra timestamp to be the current wallclock time in Cassandra.

Reply:
Just doing a put creates this state:

<img width="706" alt="screen shot 2017-12-08 at 6 10 09 pm" src="https://user-images.githubusercontent.com/20421615/33779030-1dec2694-dc43-11e7-87e2-94fa52893e30.png">

Reply:
And writing at the current timestamp i.e. after getting a fresh_timestamp, will create the following state:

<img width="576" alt="screen shot 2017-12-08 at 6 14 05 pm" src="https://user-images.githubusercontent.com/20421615/33779189-cc75505a-dc43-11e7-9fa0-430c56124c1c.png">